operator/pkg/util/util.go

```diff
@@ -145,3 +153,30 @@ func CleanFilePathJoin(root, dest string) (string, error) {
 
 	return filepath.ToSlash(newpath), nil
 }
+
+func ParseManifestStringToObjects(manifest string) (*types.ManifestResources, error) {
+	objects := &types.ManifestResources{}
+	reader := yamlUtil.NewYAMLReader(bufio.NewReader(strings.NewReader(manifest)))
+	for {
+		rawBytes, err := reader.Read()
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				return objects, nil
+			}
+
+			return nil, fmt.Errorf("invalid YAML doc: %w", err)
+		}
+
+		rawBytes = bytes.TrimSpace(rawBytes)
+		unstructuredObj := unstructured.Unstructured{}
+		if err := yaml.Unmarshal(rawBytes, &unstructuredObj); err != nil {
+			objects.Blobs = append(objects.Blobs, append(bytes.TrimPrefix(rawBytes, []byte("---\n")), '\n'))
```

Comment:
Looks dangerous, is there no inbuilt function available to write into the Blobs? If no, crazy stuff :D